distsql/distsql.go

```diff
@@ -302,7 +302,7 @@ func (pr *partialResult) Close() error {
 
 // SelectDAG sends a DAG request, returns SelectResult.
 // In kvReq, KeyRanges is required, Concurrency/KeepOrder/Desc/IsolationLevel/Priority are optional.
-func SelectDAG(goCtx goctx.Context, ctx context.Context, kvReq *kv.Request, fieldTypes []*types.FieldType) (SelectResult, error) {
+func SelectDAG(ctx context.Context, goCtx goctx.Context, kvReq *kv.Request, fieldTypes []*types.FieldType) (SelectResult, error) {
```

Comment:
should we put `goCtx` as the first parameter
PTAL @tiancaiamao

Reply:
`goCtx` should comes first, that's Go convention.

Reply:
context.Context should be the first parameter of a function (golint).
Because #1825 ?
But I fixed and passed test case, So It's ok?